src/models/room-member.ts

```diff
@@ -229,40 +224,48 @@ export class RoomMember extends TypedEventEmitter<RoomMemberEvent, RoomMemberEve
      * Update this room member's power level event. May fire
      * "RoomMember.powerLevel" if this event updates this member's power levels.
      * @param powerLevelEvent - The `m.room.power_levels` event
+     * @param roomCreateEvent - The `m.room.create` event.
+     * @param roomVersion - The room version of the room this member is in.
      *
      * @remarks
      * Fires {@link RoomMemberEvent.PowerLevel}
      */
-    public setPowerLevelEvent(powerLevelEvent: MatrixEvent): void {
+    public setPowerLevelEvent(powerLevelEvent: MatrixEvent, roomCreateEvent: MatrixEvent, roomVersion: string): void {
         if (powerLevelEvent.getType() !== EventType.RoomPowerLevels || powerLevelEvent.getStateKey() !== "") {
             return;
         }
 
         const evContent = powerLevelEvent.getDirectionalContent();
 
-        let maxLevel = evContent.users_default || 0;
         const users: { [userId: string]: number } = evContent.users || {};
-        Object.values(users).forEach((lvl: number) => {
-            maxLevel = Math.max(maxLevel, lvl);
-        });
         const oldPowerLevel = this.powerLevel;
-        const oldPowerLevelNorm = this.powerLevelNorm;
 
-        if (users[this.userId] !== undefined && Number.isInteger(users[this.userId])) {
-            this.powerLevel = users[this.userId];
-        } else if (evContent.users_default !== undefined) {
-            this.powerLevel = evContent.users_default;
-        } else {
-            this.powerLevel = 0;
+        const creators = new Set<string>();
+        if (roomVersion === "org.matrix.hydra.11" || (Number.isInteger(roomVersion) && Number(roomVersion) > 12)) {
```

Comment:
This is invalid, as per the spec https://spec.matrix.org/v1.15/rooms/

> There is no implicit ordering or hierarchy to room versions

See also https://github.com/element-hq/element-meta/issues/2921#issuecomment-3107673036

Reply:
Ah, right you are. Fixed.